sdk/digitaltwins/digitaltwins/review/digitaltwins.api.md

```diff
@@ -0,0 +1,372 @@
+    publishComponentTelemetry(digitalTwinId: string, componentPath: string, payload: string, messageId?: string): Promise<RestResponse>;
+    publishTelemetry(digitalTwinId: string, payload: any, messageId?: string): Promise<RestResponse>;
+    queryTwins(query?: string): PagedAsyncIterableIterator<any, QueryQueryTwinsResponse>;
+    updateComponent(digitalTwinId: string, componentPath: string, componentPatch: any, ifMatch?: string): Promise<DigitalTwinsUpdateComponentResponse>;
+    updateDigitalTwin(digitalTwinId: string, twinPatch: any, options?: DigitalTwinsUpdateOptionalParams): Promise<DigitalTwinsUpdateResponse>;
+    updateRelationship(digitalTwinId: string, relationshipId: string, options?: DigitalTwinsUpdateRelationshipOptionalParams): Promise<DigitalTwinsUpdateRelationshipResponse>;
+    upsertDigitalTwin(digitalTwinId: string, digitalTwinJson: string, disableUpdate?: boolean): Promise<DigitalTwinsAddResponse>;
+    upsertEventRoute(eventRouteId: string, endpointName: string, filter?: string): Promise<RestResponse>;
+    upsertRelationship(digitalTwinId: string, relationshipId: string, relationship: string, enableUpdate?: boolean): Promise<DigitalTwinsAddRelationshipResponse>;
+}
+
+// @public
+export interface DigitalTwinsDeleteMethodOptionalParams extends coreHttp.RequestOptionsBase {
+    ifMatch?: string;
+}
+
+// @public
+export interface DigitalTwinsDeleteRelationshipOptionalParams extends coreHttp.RequestOptionsBase {
+    ifMatch?: string;
+}
+
+// @public
+export interface DigitalTwinsGetByIdHeaders {
+    eTag: string;
+}
+
+// @public
+export type DigitalTwinsGetByIdResponse = DigitalTwinsGetByIdHeaders & {
+    body: any;
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: DigitalTwinsGetByIdHeaders;
+        bodyAsText: string;
+        parsedBody: any;
+    };
+};
+
+// @public
+export interface DigitalTwinsGetComponentHeaders {
+    eTag: string;
+}
+
+// @public
+export type DigitalTwinsGetComponentResponse = DigitalTwinsGetComponentHeaders & {
+    body: any;
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: DigitalTwinsGetComponentHeaders;
+        bodyAsText: string;
+        parsedBody: any;
+    };
+};
+
+// @public
+export type DigitalTwinsListIncomingRelationshipsResponse = IncomingRelationshipCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: IncomingRelationshipCollection;
+    };
+};
+
+// @public
+export interface DigitalTwinsListRelationshipsOptionalParams extends coreHttp.RequestOptionsBase {
+    relationshipName?: string;
+}
+
+// @public
+export type DigitalTwinsListRelationshipsResponse = RelationshipCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: RelationshipCollection;
+    };
+};
+
+// @public
+export interface DigitalTwinsSendComponentTelemetryOptionalParams extends coreHttp.RequestOptionsBase {
+    dtTimestamp?: string;
```

Comment:
`dtTimestamp` doesn't seem super descriptive to me.

Reply:
We discussed: this comes from the swagger and we have a work item to improve this name for ADT GA.

---
In reply to: [455324864](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r455324864) [](ancestors = 455324864)

Reply:
FWIW, in the .NET client, we called this TimeStamp for now, and will update it along with the swagger change at GA.

Reply:
Added to the swagger changes list

---
In reply to: [455388544](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r455388544) [](ancestors = 455388544,455324864)